specs/app.spec.js

```diff
@@ -66,49 +54,104 @@ describe('When the user has csv files in the files folder', function () {
-                if (error) {
-                    done(error);
-                }
-                expect(data.length).to.deep.equal(0);
-                done();
-            });
+
+            connectAndQuery('SHOW TABLES LIKE \'t1\'')
+                .then(data => {
+                    expect(data.length).to.deep.equal(0);
+                    done();
+                });
         });
 
         it('Should have not inserted t2 csv in the files folder', function (done) {
-            var connection = mysql.createConnection(DB_CONFIG);
-            connection.query('SHOW TABLES LIKE \'t2\'', (error, data) => {
-                connection.end();
-                if (error) {
-                    done(error);
-                }
-                expect(data.length).to.deep.equal(0);
-                done();
-            });
+
+            connectAndQuery('SHOW TABLES LIKE \'t2\'')
+                .then(data => {
+                    expect(data.length).to.deep.equal(0);
+                    done();
+                });
+        });
+    });
+
+    describe('When the user use the flag --append', function () {
+        beforeEach(function () {
+            return runCommand('node app.js');
+        });
+
+        beforeEach(function () {
+            return runCommand('node app.js --append');
+        });
+
+        it('Should have appended the t1 csv as expected', function (done) {
+            connectAndQuery('SELECT * FROM t1')
+                .then(data => {
+                    expect(data).to.deep.equal([
+                        { '"id"': '1', 'first name': '"SMH"', 'person\'s email': 'smh@email.com' },
+                        { '"id"': '2', 'first name': 'MO\'', 'person\'s email': 'mo@email.com' },
+                        { '"id"': '3', 'first name': 'hello, world', 'person\'s email': 'someemail@email.com' },
+                        { '"id"': '1', 'first name': '"SMH"', 'person\'s email': 'smh@email.com' },
+                        { '"id"': '2', 'first name': 'MO\'', 'person\'s email': 'mo@email.com' },
+                        { '"id"': '3', 'first name': 'hello, world', 'person\'s email': 'someemail@email.com' }
+                    ]);
+                    done();
+                });
         });
 
+        it('Should have appended the t2 csv as expected', function (done) {
+            connectAndQuery('SELECT * FROM t2')
+                .then(data => {
+                    expect(data).to.deep.equal([
+                        { COL1: '"Hello, World!"', COL2: '```' },
+                        { COL1: '"Hello, World!"', COL2: '```' }
+                    ]);
+                    done();
+                });
+        });
+    });
+
+    describe('When the user runs the app.js using the flag --overwrite', function () {
+        beforeEach(function () {
+            return runCommand('node app.js');
+        });
+
+        beforeEach(function () {
+            return runCommand('node app.js --overwrite --files t1.csv');
```

Comment:
It's not good practice to have 2 beforeEach registrations at the same level of nesting. You should do this instead:
```
return runCommand(command1)
    .then(() => runCommand(command2));